### DISCUSSION TITLE
Should setting the top level `kind='floating'` automatically apply to `commit_popup` and other child processes of Neogit?

### DISCUSSION THREAD

**Discussion by lessej** [NeogitOrg/neogit#699] [Ideas]
Hello, I chose to put this question in discussion because I'm unsure if it's intended behavior or not.  If so, my usage of Neogit must be different than intended so I would be grateful if you could steer me back to the right course.  Thanks!

### Overview
When I was updating my Neogit configuration to use a floating window instead of splitting, I noticed that setting `kind='floating'`  will open Neogit in a floating window but popups that can be opened from within Neogit will continue to open as splits on the window from which Neogit was originally opened.  The Neogit floating window often partially obscures the split window (for commit, push/pull, etc.) which makes it hard to read.  Additionally, I have not found a good way to close the original Neogit window (although this may be down to me not knowing the right Neovim commands).

I also noticed that despite setting `commit_popup` to floating, the commit message window does not open in a popup.  Please see the following screenshots:

#### Example 1:
```
require('neogit').setup{
    kind = 'floating'
}
```
<img width="1355" alt="Screenshot 2023-08-04 at 16 33 11" src="https://github.com/NeogitOrg/neogit/assets/44124944/bc0f359f-1463-40fb-8103-689fae4afd45">

#### Example 2:
```
require('neogit').setup{
    kind = 'floating',
    popup = {
        kind = 'floating'
    },
    commit_popup = {
        kind = 'floating'
    },
    preview_buffer = {
        kind = 'floating'
    }
}
```
<img width="1236" alt="Screenshot 2023-08-04 at 17 16 47" src="https://github.com/NeogitOrg/neogit/assets/44124944/75c4c84d-c873-4a11-93aa-a23bc274a22e">

### Possible solution
When considering how I use Neogit, I think the following solutions would feel smoother:
1. If `kind='floating'` is set on the top level, all popups than can be opened from within Neogit are also set to `'floating'` so that they open within/on top of the open Neogit floating window.
2. If `kind='floating'` is set on the top level, split the Neogit window rather than the window from which is was opened.

I'm sorry if this is a duplicate issue or if I'm misunderstanding the intended behavior!

**Comment by CKolkey** [comment]
Hey! I think off the bat, I'll just say that the `"floating"` implementation isn't as good as it could be. Of all the work we've done here over the last half year, precisely none of it has been on that. Your description of the intended behaviour makes perfect sense to me.. Thing is, I don't use the floating windows, and IMO all the extra logic for window management is a PITA that I don't want to deal with (though I'm open to PR's). 

The "floating" implementation is considered experimental, and as you illustrated, very much lives up to that. For a better experience, I recommend using a `tab` instead :)

**Reply by lessej** [reply]
Ah, understood!  I wasn't aware `floating` was an experimental feature.  I don't have much experience writing Neovim plugins but I quite like using the floating window so perhaps I'll give the remaining implementation a go.  Thanks for the response!

**Reply by CKolkey** [reply]
Glad I could help :)
